contracts/Gate.sol

```diff
@@ -162,4 +169,130 @@ contract Gate is ProtocolVersioned, Owned {
+    function acceptStakeRequest(address _staker, bytes32 _hashLock)
+        external
+        returns (bool /* success */)
+    {
+        // check if the caller is whitelisted worker
+        //require(workers.isWorker(msg.sender)); //TODO: revist this to add worker check
+
+        StakeRequest storage stakeRequest = stakeRequests[_staker];
+
+        // check if the stake request exists
+        require(stakeRequest.beneficiary != address(0));
+
+        // check if the stake request was not accepted
+        require(stakeRequest.hashLock == bytes32(0));
+
+        // check if _hashLock is not 0
+        require(_hashLock != bytes32(0));
+
+        // Transfer bounty amount form worker to gate contract
+        require(OpenSTValueInterface(openSTProtocol).valueToken().transferFrom(msg.sender, address(this), bounty));
+
+        // Approve OpenSTValue contract for stake amount
+        require(OpenSTValueInterface(openSTProtocol).valueToken().approve(openSTProtocol, stakeRequest.amount));
+
+
+        uint256 amountUT = 0;
+        uint256 nonce = 0;
+        uint256 unlockHeight = 0;
+        bytes32 stakingIntentHash = bytes32(0);
+
+        (amountUT, nonce, unlockHeight, stakingIntentHash) = OpenSTValueInterface(openSTProtocol).stake(
+            uuid,
+            stakeRequest.amount,
+            stakeRequest.beneficiary,
+            _hashLock,
+            _staker);
+
+        // Check if the stake function call did not result in to error.
+        require(stakingIntentHash != bytes32(0));
+
+        stakeRequests[_staker].unlockHeight = unlockHeight;
+        stakeRequests[_staker].hashLock = _hashLock;
+
+        emit StakeRequestAccepted(_staker, stakeRequest.amount, amountUT, nonce, unlockHeight, stakingIntentHash);
+
+        return true;
+    }
+
+
+  function processStaking(
+    bytes32 _stakingIntentHash,
+    bytes32 _unlockSecret)
+    external
+    returns (bool /* success */)
+  {
+    // check if the caller is whitelisted worker
+    //require(workers.isWorker(msg.sender)); //TODO: revist this to add worker check
+
+    require(_stakingIntentHash != bytes32(0));
+
+    address staker = OpenSTValueInterface(openSTProtocol).getStakerAddress(_stakingIntentHash);
+
+    StakeRequest storage stakeRequest = stakeRequests[staker];
+
+    // check if the stake request exists
+    require(stakeRequest.beneficiary != address(0));
+
+    // check if the stake request was accepted
+    require(stakeRequest.hashLock != bytes32(0));
+
+    // we call processStaking for OpenSTValue and get the stakeAddress on success.
+    address stakerAddress = OpenSTValueInterface(openSTProtocol).processStaking(_stakingIntentHash, _unlockSecret);
+
+    // check if the stake address is not 0
+    require(stakerAddress != address(0));
+
+    // Transfer bounty amount to the msg.sender account
+    require(OpenSTValueInterface(openSTProtocol).valueToken().transfer(msg.sender, bounty));
```

Comment:
I've mentioned this in the general comment, but here we want to check that if `isWorker(msg.sender)` then transfer it to `workers`, the contract address, otherwise transfer to `msg.sender` (so as the Workers check is not yet enforced, this line is consistent, but best to already include the condition)

Reply:
Added the commented code with TODO.